plugins/modules/katello_repository_set.py

```diff
@@ -197,9 +199,13 @@ def main():
     current_repo_names = set(map(lambda repo: repo['name'], current_repos))
     desired_repo_names = set(map(lambda repo: repo['repo_name'], desired_repos))
 
+    if not desired_repo_names:
+        module.warn("Desired repositories not found for repository set {0}. Available: {1}"
```

Comment:
I am unsure if this should be a `warn` or `fail`.

Reply:
cc @akofink @sean797

Reply:
IMO it should fail because the module cannot move forward. Correct me if I'm wrong, but it looks like if they specify 3 repos and only 1 is found, the module will continue like normal. Maybe that should be where the warning happens - the module can do part of the work requested.

Reply:
We handle the "asked for 3, but only 1 found" case in the next lines:
https://github.com/theforeman/foreman-ansible-modules/blob/aaa72556db39ac82b692da69df4a5e083bab8f0a/plugins/modules/katello_repository_set.py#L200-L202

Reply:
Ah, okay :+1: So it seems users would get two similar messages. I don't understand why line 200 isn't sufficient to handle both cases?

Reply:
This tries to handle two different (but related, in a way) cases:
1. the original check is "the user requested more than we can provide"
1. the check I added is "the user didn't request anything (that we could find)"

We could fold both `if` statements into one and just yield one error message

Reply:
I think that's the best option, especially since the two error messages are essentially the same.

Reply:
However, now that I've said this, I think the old check was never called, because `desired_repos` is calculated from `available_repos`, so it can never be bigger than that.

Reply:
Basically if `desired_repos is not available_repos` we want this error - I'm fine with using this PR to fix this other discovered issue.

Reply:
did you mean "desired is strict subset of available"? because desired can very well be not available when you have 7.7, 7.6 and 7.5 provided as available, but you only want to enable 7.7.

Reply:
As we said before, desired will always be a strict subset of available, no?

Reply:
"hah"